apps/examples/websocket/websocket_main.c

```diff
@@ -747,9 +749,17 @@ int websocket_main(int argc, char *argv[])
 			goto error_with_input;
 		}
 
-		strncpy(input->server_ip, argv[2], 20);
-		strncpy(input->server_port, argv[3], 8);
-		strncpy(input->path, argv[4], 32);
+		int addr_len = strlen(argv[2]);
+		int port_len = strlen(argv[3]);
+		int path_len = strlen(argv[4]);
+
+		if (addr_len > WEBSOCKET_EXAMPLE_ADDR_LEN || port_len > WEBSOCKET_EXAMPLE_PORT_LEN
+			|| path_len > WEBSOCKET_EXAMPLE_PATH_LEN) {
+			goto error_with_input;
+		}
+		strncpy(input->server_ip, argv[2], addr_len + 1);
+		strncpy(input->server_port, argv[3], port_len + 1);
+		strncpy(input->path, argv[4], path_len + 1);
```

Comment:
If you put addr_len + 1 to add '\0', you had better use snprintf instead of strncpy I think.

Reply:
I think strncpy is also ok.